### PR TITLE
Fix according to atomicassets usage

### PR DESCRIPTION
**Pull request by apporc**
1. added declare keyword for variant type value field.
2. Fixed variant interface, please note a variant is not a union type, it is a struct.
3. Fixed losing array decorator while following type alias
4. Added makefile command to update all mock contracts
5. Updated all mock contracts

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -22,6 +22,16 @@ ci-test: node_modules
 test_generate: node_modules clean lib
 	node lib/cli.js generate eosio.token -u https://jungle4.greymass.com
 
+.PHONY: update_mock_contracts
+update_mock_contracts: node_modules clean lib
```
Comment (dafuga) [inline Makefile:26]: 💯

File: src/commands/contract/finders.ts [hunk 2]
```diff
@@ -17,10 +17,10 @@ export const ANTELOPE_CLASS_MAPPINGS = {
 export const ANTELOPE_CLASS_WITHOUT_TYPES = ['BlockTimestamp', 'TimePointSec']
 
 export function findTypeFromAlias(typeString: string, abi: ABI.Def): string | undefined {
-    const {type: typeStringWithoutDecorator, decorator} = extractDecorator(typeString)
+    const {type: typeStringWithoutDecorator} = extractDecorator(typeString)
     const alias = abi.types.find((type) => type.new_type_name === typeStringWithoutDecorator)
 
-    return alias?.type && `${alias?.type}${decorator || ''}`
```
Comment (dafuga) [inline src/commands/contract/finders.ts:23]: I'm not sure about this 🤔 Wouldn't we want to either return the decorator here or use it to find the type on line 21? Do you have an example of some code that doesn't work with the current code?

File: test/data/contracts/mock-atomicassets.ts [hunk 3]
```diff
@@ -103,31 +103,33 @@ export namespace ActionParams {
         }
         export interface pair_string_ATOMIC_ATTRIBUTE {
             key: string
-            value: Types.pair_string_ATOMIC_ATTRIBUTE_value_variant
+            value: Types.variant_int8_int16_int32_int64_uint8_uint16_uint32_uint64_float32_float64_string_INT8_VEC_INT16_VEC_INT32_VEC_INT64_VEC_UINT8_VEC_UINT16_VEC_UINT32_VEC_UINT64_VEC_FLOAT_VEC_DOUBLE_VEC_STRING_VEC
+        }
+        export interface variant_int8_int16_int32_int64_uint8_uint16_uint32_uint64_float32_float64_string_INT8_VEC_INT16_VEC_INT32_VEC_INT64_VEC_UINT8_VEC_UINT16_VEC_UINT32_VEC_UINT64_VEC_FLOAT_VEC_DOUBLE_VEC_STRING_VEC {
+            value:
```
Comment (dafuga) [inline test/data/contracts/mock-atomicassets.ts:109]: This feels pretty unintuitive 😅 I'm guessing that the current code doesn't work with some variants on the atomicassets contract? I wonder if there is a way to have the contract automatically wrap the variant value in the correct format 🤔

File: src/commands/contract/structs.ts [hunk 4]
```diff
@@ -171,13 +171,16 @@ export function generateField(
...
-    const isArray = field.type.endsWith('[]')
+    let isArray = field.type.endsWith('[]')
 
     // Start with the main type argument
     const decoratorArguments: (ts.ObjectLiteralExpression | ts.StringLiteral | ts.Identifier)[] = [
         findFieldStructType(field.type, namespace, abi),
     ]
 
+    const structTypeString = findFieldStructTypeString(field.type, namespace, abi)
+    isArray = isArray || structTypeString.endsWith('[]')
```
Comment (dafuga) [inline src/commands/contract/structs.ts:182]: Good catch 👍

**Review by aaroncox — APPROVED** [review]

**Pull request merged**